### PR TITLE
[Community Feedback Required] Added technical vision for Web API components

### PR DESCRIPTION
**Pull request by paliarush**
Preview of the technical vision is available here https://github.com/paliarush/devdocs/blob/webapi-technical-vision/guides/v2.0/coding-standards/technical-vision/webapi.md .
Please note, that references to devdocs may be broken (compilation is required).

Feel free to leave your comments in the PR.

### CODE REVIEW

File: guides/v2.0/coding-standards/technical-vision/webapi.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+Web API is crucial for Magento because of the need to integrate with order management, customer management and other enterprise management software systems.
+
+There are a lot of headless magento installations, where Magento functionality is used only partially, while the other pieces of eCommerce website are provided by other systems. 
+
+See [more details](https://en.wikipedia.org/wiki/Web_API) about importance of web APIs in modern web applications.
+
+
+### Components Dependencies
+Below are some basic examples of what you can do with markdown.
```
Comment (navarr) [inline guides/v2.0/coding-standards/technical-vision/webapi.md]: I think this is text from a template accidentally left in

**Comment by navarr** [comment]
Some confusion for me revolves around the Service Contracts / WebAPI.

I know the MSI team is separating out their API into a good bit of modules: Inventory, InventoryApi, and now separate additional modules for the AdminUi and FrontendUi.  The graphics in this PR keep the API in the same initial module.

Is MSI an outlier or is it not necessary to show the service contracts belonging to their own module?

File: guides/v2.0/coding-standards/technical-vision/webapi.md (hunk 1 above)
Comment (paliarush) [reply guides/v2.0/coding-standards/technical-vision/webapi.md]: Good point, thanks

**Comment by paliarush** [comment]
@navarr current technical vision is focused mostly on the place of Web APIs. There is separate technical vision being prepared for Service Contracts, it reflects having separate modules for APIs. It will be published after we finalize vision for query API (work in progress and will be integrated with GraphQL)

**Comment by keharper** [comment]
Checked in the final Web API Technical Vision article under

https://github.com/magento/devdocs_internal/pull/1746

Closing this one.

**Pull request closed without merging**